ql/quotes/compositequote.hpp

```diff
@@ -41,7 +41,7 @@ namespace QuantLib {
         CompositeQuote(
             const Handle<Quote>& element1,
             const Handle<Quote>& element2,
-            const BinaryFunction& f);
+            const BinaryFunction& f = BinaryFunction());
```

Comment:
Does this work when the function doesn't have a default constructor?

Reply:
For me yes (clang 6.0.0):

```
#include <ql/quotes/derivedquote.hpp>
#include <ql/quotes/simplequote.hpp>

#include <boost/make_shared.hpp>

using namespace QuantLib;

class Fct {
public:
    Fct() = delete;
    Fct(const int) {}
    Real operator()(const Real x) const { return x; }
};

Handle<Quote> q(boost::make_shared<SimpleQuote>(0.0));

Handle<Quote> d1(boost::make_shared<DerivedQuote<Fct>>(q));          // compile error
Handle<Quote> d2(boost::make_shared<DerivedQuote<Fct>>(q, Fct(42))); // ok

int main() { return 0; }
```